HIPs/incubation/web3j.md

```diff
@@ -0,0 +1,204 @@
+
+## Core Web3j library
+
+The [core Web3j library](https://github.com/web3j/web3j) was designed to be highly modular, so that users could easily incorporate the minimal parts of the library for their projects. This consideration was especially important in servicing the Android community.
+
+The notable modules include:
+
+- **rlp:** [Recursive Length Prefix (RLP)](https://ethereum.org/en/developers/docs/data-structures-and-encoding/rlp/) serialisation
+- **crypto:** cryptographic operations and key management 
+- **abi:** [Contract Application Binary Interface (ABI)](https://docs.soliditylang.org/en/latest/abi-spec.html) encoding and decoding
+- core: core Web3j components including the JSON-RPC implementation and abstractions for wallets
+- utils: utility functions including type conversions
+- **besu, geth, eea: Ethereum** client specific functionality
+- codegen: code generators for creating Java type wrappers for Solidity smart contracts
+- contracts: smart contract specific types
+
+## Smart Contract Testing
+
+The Web3j core library is complemented by the following projects that facilitate unit and integration testing of Solidity smart contracts.
+
+The [Web3j EVM](https://github.com/web3j/web3j-evm) project provides an embedded EVM that allows developers to write applications that simulate transactions taking place on the EVM. 
+
+The EVM contained within the project is the Hyperledger Besu EVM. This ensures that the EVM can be run within the local JVM without the need to orchestrate a full Ethereum node via Docker or similar tools.
+
+One of the significant advantages of using the Hyperledger Besu EVM is that developers work with a production-grade EVM during the development process. In other non-JVM development environments, it's unlikely that developers will have at their disposal an embedded EVM that uses the same code as a widely used Ethereum client.
+
+The Web3j EVM project is complemented by the [Web3j Unit](https://github.com/web3j/web3j-unit) project. This project is used to inject different EVM environments into unit and integration tests.
+
+This allows developers to transact with real Ethereum execution environments from their unit tests. The most common use cases are smart contract deployments and interacting with them, but any type of EVM transaction can be performed.
+
+By default the Web3j unit project uses the embedded Hyperledger Besu EVM for this purpose to ensure fast execution time. However, it is also possible to use containerised instances of Ethereum execution clients such as Hyperledger Besu and Geth for this purpose. This is especially useful for integration testing purposes to simulate production environments.
+
+## Build Plugins
+
+The JVM platform utilised two major build systems — Gradle and Maven. The [Web3j Gradle](https://github.com/web3j/web3j-gradle-plugin) and [Web3j Maven](https://github.com/web3j/web3j-maven-plugin) projects implement support for these build systems.
```

Comment:
Would these two repositories also be coming to Hyperledger?

Reply:
Yes.